algoliasearch/transport.go

```diff
@@ -86,7 +87,7 @@ func NewTransportWithHosts(appId, apiKey string, hosts []string) *Transport {
 func defaultHeaders(appId, apiKey string) map[string]string {
 	return map[string]string{
 		"Connection":               "keep-alive",
-		"User-Agent":               "Algolia for Go (" + version + ")",
+		"User-Agent":               "Go ("+ runtime.Version()[2:] +"); Algolia for Go (" + version + ")",
```

Comment:
1. We don't want to panic if the string is shorter than 2 characters long. Why don't we keep the full runtime.Version() instead, which makes it complete, as @maxiloc suggested on Slack. It would let us use it with more flexibility on our side.

2. For clarity sake, let's use `fmt.Sprintf` to properly format the User-Agent here, such as:
```go
fmt.Sprintf("Go (%s); Algolia for Go (%s)", runtime.Version(), version)
```

Reply:
I'm not sure we'll ever need the first 2 letter in `go1.10.1` but we definitely don't want to panic! I'm updated 👌